napari/layers/labels/labels.py

```diff
@@ -784,7 +775,13 @@ def paint(self, coord, new_label, refresh=True):
             sphere_dims = len(coord)
             # Ensure circle doesn't have spurious point
             # on edge by keeping radius as ##.5
-            radius = np.floor(self.brush_size / 2) + 0.5
+            radius = (
+                np.floor(
+                    np.mean([brush_size_dims[d] for d in self._dims.displayed])
+                    / 2
+                )
+                + 0.5
+            )
```

Comment:
Can you maybe expand the comment above to explain what is happening here?

Reply:
comment added

Reply:
@sofroniewn I probably should have expanded on my question: I don't understand why we've gone from a single size to the mean of several sizes — which seems very brittle. Indeed, following the code above I would say that this is incorrect — in this case in data coordinates we would want an elliptical brush, right? We can merge as-is (our out-of-plane annotation has never been great anyway), **but**, the comment should reflect that this is a huge hack. 😂

Reply:
> in this case in data coordinates we would want an elliptical brush, right? We can merge as-is (our out-of-plane annotation has never been great anyway), but, the comment should reflect that this is a huge hack. 😂

Correct, we don't support that yet, and never have. Problem is definitely not worse now, it's always been an issue, but I will update comment